### ISSUE TITLE
Attached resource update doesn't filter out readonly/hidden fields

### ISSUE THREAD

**Issue by Tarpsvo** [laravel/nova-issues#4787]
- Laravel Version: 9.28.0
- Nova Version: 4.13.0
- PHP Version: 8.1.9
- Database Driver & Version: MySQL 8

### Description:

When using dependsOn fields in a pivot setting, the readonly fields are not filtered out before calling ->fill() on them.

When looking at FillsFields, functions `fill` and `fillForUpdate` filter our readonly fields (`->withoutReadonly()`).

However, their pivot counterparts `fillPivot` and `fillPivotForUpdate` do not filter our the readonly fields.

I think they should act the same and both filter out readonly fields.

**Comment by crynobone** [comment]
Can you provide a reproducing repository to demonstrate the issue?

**Comment by Tarpsvo** [comment]
I don't think it's necessary. I will instead provide you with a screenshot, cause they say a picture says more than a thousand words.

![image](https://user-images.githubusercontent.com/2018660/189112516-068d1bac-4334-40fb-9a25-0664df3beea7.png)

The motivation for this change is to have a way to filter out pivot fields **after** dependsOn() is applied so their `->fill()` functions aren't fired.

**Comment by crynobone** [comment]
Yes, I know there is a difference. **However**, I cannot make an example (tried and it does work fine) where we can create a dusk test to show that it's failing at the moment and a fix can be introduced to solve it. 

I can make a PR to just add those lines but it has a high chance where it gets rejected without:

- reproducing dusk/feature/integration tests
- I myself would need to be able to understand and answers any question from code reviewer.

**Comment by Tarpsvo** [comment]
My usecase was fairly complicated.

I have a Select which some fields depend on. The select allows the user to toggle between "Manual value" and "Existing value" inputs.

If the user selects "Manual" input, they're shown the following fields:
Text::make('Value')
Text::make('Filter value')

However, if the user selects "Existing value", I render a Select which allows the user to select between existing values. The select in Nova uses a `fillUsing` override which fills in the 'value' and 'filter_value' automatically.

This creates a conflict where two fields compete for filling in the value. So I was faced with finding a way to stop fields from being passed to the `fillFields` function. When digging into the code, I noticed that readonly fields (after applying dependsOn) are skipped when doing non-pivot updates. So that would enable me to fill in the data without any fear of conflicts appearing.

My current solution is to just check the state of the primary "Manual"/"Existing" select from the request inside the "Existing value" Select's fillUsing method and stop it from executing if it doesn't equal "existing".

I just thought it might be nice to keep things consistent and filter out readonly fields in both cases (non-pivot and pivot), so it would be consistent.
